### ISSUE TITLE
Open Question: Incentivization of PoW verification (Proof-of-proof-of-work)

### ISSUE THREAD

**Issue by christobias** [unprll-project/unprll#5]
**Problem:** Since the Unprll Proof-of-work is expensive to verify, sync-up is slow. Hence, shifting verification to the miner network and giving them another chance to earn ULL is a good idea. Since verification can occur in parallel, multithreaded miners can have a chance without having to resort to multiple wallets, further keeping single core mining competitive.

This thread is open for discussion on implementing such a feature

**Comment by RealAwesomeness** [comment]
I say we do something similar to masternodes. However, I feel like that conflicts with a vision of 1 cpu 1 voice because people with more coin will hold more power.

**Comment by christobias** [comment]
Yes, ideally I'd make it similar to mining and keep it a competition between miners

**Comment by noxsphere** [comment]
idea => https://blog.turtlecoin.lol/archives/running-a-public-node-for-fun-profit/

**Comment by noxsphere** [comment]
misread oupsss ;)

**Comment by ovsiannikov** [comment]
something like this
https://www.veriblock.org/wp-content/uploads/2018/03/PoP-White-Paper.pdf

**Comment by christobias** [comment]
That's an interesting idea, but that's for inheriting proof-of-work from other chains onto your current chain :)

**Comment by christobias** [comment]
I have a potential idea for this. Ideally an algorithmic solution is preferable, but if that can't be done, a threat-based approach could be taken. Suppose a verifier wishes to get paid for verifying proofs-of-work. They will have to pledge a certain amount of his current balance as a "security deposit". Then, after a set number of blocks have passed without any invalid proofs, the verifier gets paid his original deposit and an extra amount for his services (Possibly more for any invalid blocks they did find during that timeframe). In case an invalid proof was seen (and it wasn't at the highest block at that time), all deposits are confiscated/burned.

Opinions and critique welcomed :)

**Comment by RealAwesomeness** [comment]
That would benefit people who purposely broadcast invalid hashes to themselves to gain more coin.

**Comment by christobias** [comment]
Hmm, so extra payment for invalid blocks is counterproductive.
